mwdb/web/src/components/OAuth.js

```diff
@@ -0,0 +1,133 @@
+import { View, getErrorMessage } from "@mwdb-web/commons/ui";
+
+export function OAuthLogin() {
+    const api = useContext(APIContext);
+    const [error, setError] = useState();
+    const [providers, setProviders] = useState([]);
+    const [chosenProvider, setChosenProvider] = useState();
+
+    async function getProviders() {
+        try {
+            const response = await api.axios.get("/oauth");
+            setProviders(response.data["providers"]);
+        } catch (e) {
+            setError(e);
+        }
+    }
+
+    async function login(provider) {
+        try {
+            const response = await api.axios.post(
+                `/oauth/${provider}/authenticate`
+            );
+            let expirationTime = new Date();
+            expirationTime.setTime(expirationTime.getTime() + 5 * 60 * 1000);
+            sessionStorage.setItem(
+                `openid_${response.data["state"]}`,
+                JSON.stringify({
+                    provider: provider,
+                    nonce: response.data["nonce"],
+                    action: "login",
+                    expiration: expirationTime,
+                })
+            );
+            window.location = response.data["authorization_url"];
+        } catch (e) {
+            setError(e);
+        }
+    }
+
+    useEffect(() => {
+        getProviders();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <View error={error}>
+            <form
+                onSubmit={(e) => {
+                    e.preventDefault();
+                    login(chosenProvider);
+                }}
+            >
+                <select
+                    className="custom-select"
+                    value={chosenProvider}
+                    onChange={(ev) => setChosenProvider(ev.target.value)}
+                >
+                    <option value="" hidden>
+                        Select provider...
+                    </option>
+                    {providers.map((provider) => (
+                        <option value={provider}>{provider}</option>
+                    ))}
+                </select>
+                <button type="submit" className="btn btn-outline-success">
+                    Login
+                </button>
+            </form>
+        </View>
+    );
+}
+
+export function OAuthAuthorize() {
+    const api = useContext(APIContext);
+    const auth = useContext(AuthContext);
+    const history = useHistory();
+    // Current query set in URI path
+    const { code, state } = queryString.parse(history.location.search);
+
+    async function authorize() {
+        const stateData = sessionStorage.getItem(`openid_${state}`);
+        if (!stateData) {
+            history.push("/", { error: "Invalid state data" });
+        }
+        let { provider, nonce, action, expiration } = JSON.parse(stateData);
+        try {
+            expiration = Date.parse(expiration);
```

Comment:
If we use epoch (number)
```suggestion
            expiration = new Date(expiration);
```